spec/std/int_spec.cr

```diff
@@ -583,8 +603,8 @@ describe "Int" do
   end
 
   it "returns 0 when doing IntN::MIN % -1 (#8306)" do
-    {% for n in [8, 16, 32, 64] %}
-      (Int{{n}}::MIN % -1_i{{n}}).should eq(0)
+    {% for n in [8, 16, 32, 64, 128] %}
+      (Int{{n}}::MIN % -1.to_i{{n}}).should eq(0)
```

Comment:
What's the reason for these transformations? Shouldn't `-1_i128` work as well?

Reply:
I need to test that
Was unsure because of #7915

Reply:
No, this is required
Using -1_i128 raises a compile-time exception (OverflowError)

This needs to be changed in part 2

Reply:
ah, then it doesn't close #7915 as mentioned in the description. thanks for testing

Reply:
> ah, then it doesn't close #7915 as mentioned in the description. thanks for testing

This PR does close #7915, but the std_specs need to be successful with the "old" compiler for this PR to be accepted. Once this PR was merged, it can be changed back after the first crystal release that includes this PR.